operator/v1/types_ingress.go

```diff
@@ -0,0 +1,119 @@
+	// specified, this value:
+	//
+	// * Is added automatically to the status of Routes to inform downstream systems.
+	// * May be used for a wildcard DNS alias to any provisioned load balancer.
+	// * May be used to generate a default wildcard certificate.
+	//
+	// If nil, the default value from the cluster ingress config is used.
+	//
+	// Once set, this field cannot be changed.
+	IngressDomain *string `json:"ingressDomain"`
+
+	// Replicas is the desired number of router instances.
+	Replicas int32 `json:"replicas"`
+
+	// HighAvailability describes the kind of high availability mechanism to apply
+	// to the cluster ingress. If nil, a default appropriate for the
+	// infrastructure platform is used.
+	//
+	// Once set, this field cannot be changed.
+	HighAvailability *ClusterIngressHighAvailability `json:"highAvailability"`
+
+	// DefaultCertificateSecret is the name of a secret containing the default
+	// certificate configuration. The secret is assumed to contain:
+	//
+	// * tls.crt
+	// * tls.key
+	DefaultCertificateSecret *string `json:"defaultCertificateSecret"`
+
+	// NamespaceSelector is used to filter the set of namespaces serviced by the
+	// router.
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector"`
+
+	// RouteSelector used to filter the set of routes serviced by the router.
+	RouteSelector *metav1.LabelSelector `json:"routeSelector"`
+
+	// NodePlacement enables explicit control over the scheduling of the router.
+	NodePlacement *NodePlacement `json:"nodePlacement"`
+}
+
+// NodePlacement describes node scheduling configuration for routers.
+type NodePlacement struct {
+	// NodeSelector is the node selector applied to router deployments. By
+	// default, routers are scheduled only on worker nodes (i.e. nodes labelled
+	// with `node-role.kubernetes.io/worker`).
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector"`
+}
+
+type ClusterIngressHAType string
+
+const (
+	// CloudClusterIngressHA configures container-networked routers and exposes
+	// them with a Service Load Balancer.
+	// https://kubernetes.io/docs/concepts/services-networking/#loadbalancer
+	CloudClusterIngressHA ClusterIngressHAType = "Cloud"
+
+	// UserDefinedClusterIngressHA configures router deployments to use host
+	// networking, exposing node ports 80 and 443. Users are responsible for
+	// connecting the routers to a load balancer.
+	UserDefinedClusterIngressHA ClusterIngressHAType = "UserDefined"
+
+	// NoClusterIngressHA configures container-networked routers. Users are
+	// responsible for exposing the routers or connecting them to a load balancer.
+	NoClusterIngressHA ClusterIngressHAType = "None"
+)
+
+// ClusterIngressHighAvailability is the high availability configuration for a
+// ClusterIngress.
+type ClusterIngressHighAvailability struct {
+	Type ClusterIngressHAType `json:"type"`
+}
+
+// ClusterIngressStatus defines the observed status of the ClusterIngress.
+type ClusterIngressStatus struct {
```

Comment:
Add conditions and make sure you have an `Available` condition (at minimum).  I think that's the bar for GA

Reply:
Should we be using `OperatorStatus` here or rolling our own?

Reply:
@smarterclayton still need to know what to do here (define our own conditions vs. embedding some common struct)